### PR TITLE
K8s: Add tracing to apiserver startup

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

Adds generic tracing for dskit service startup so that it can be used with apiserver startup.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/services/apiserver/service.go [hunk 1]
```diff
@@ -86,7 +89,7 @@ type RestConfigProvider interface {
 
 type DirectRestConfigProvider interface {
 	// GetDirectRestConfig returns a k8s client configuration that will use the same
-	// logged logged in user as the current request context.  This is useful when
+	// tged logged in user as the current request context.  This is useful when
```
Comment (charandas) [inline pkg/services/apiserver/service.go:92]: tged?

Comment (toddtreece) [reply pkg/services/apiserver/service.go:92]: no idea. probably my daughter 🤣

Comment (stephaniehingtgen) [inline pkg/services/apiserver/service.go:92]:
```suggestion
	// logged in user as the current request context.  This is useful when
```

**Review by stephaniehingtgen — APPROVED** [review]
One small change, looks great! Thank you!

**Pull request merged**